src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskCompleteByIdControllerTest.java

```diff
@@ -0,0 +1,418 @@
+            CamundaTask camundaTasks = mockServices.getCamundaTask("processInstanceId", taskId);
+            when(camundaServiceApi.getTask(any(), eq(taskId))).thenReturn(camundaTasks);
+
+            doThrow(FeignException.FeignServerException.class)
+                .when(camundaServiceApi).addLocalVariablesToTask(any(), any(), any());
+
+            mockMvc.perform(
+                post(ENDPOINT_BEING_TESTED)
+                    .header(AUTHORIZATION, IDAM_AUTHORIZATION_TOKEN)
+                    .header(SERVICE_AUTHORIZATION, SERVICE_AUTHORIZATION_TOKEN)
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+            ).andExpect(
+                ResultMatcher.matchAll(
+                    status().is5xxServerError(),
+                    content().contentType(APPLICATION_PROBLEM_JSON_VALUE),
+                    jsonPath("$.type").value("https://github.com/hmcts/wa-task-management-api/problem/task-complete-error"),
+                    jsonPath("$.title").value("Task Complete Error"),
+                    jsonPath("$.status").value(500),
+                    jsonPath("$.detail").value(
+                        "Task Complete Error: Task complete failed. Unable to update task state to completed.")
+                ));
+        }
+
+        @Test
+        void should_return_500_with_application_problem_response_when_complete_call_fails() throws Exception {
+
+            mockServices.mockServiceAPIs();
+
+            when(permissionEvaluatorService.hasAccessWithAssigneeCheckAndHierarchy(any(), any(), any(), any(), any()))
+                .thenReturn(true);
+
+            CamundaTask camundaTasks = mockServices.getCamundaTask("processInstanceId", taskId);
+            when(camundaServiceApi.getTask(any(), eq(taskId))).thenReturn(camundaTasks);
+
+            doNothing().when(camundaServiceApi).addLocalVariablesToTask(any(), any(), any());
+            doThrow(FeignException.FeignServerException.class).when(camundaServiceApi).completeTask(any(), any(), any());
+
+            mockMvc.perform(
+                post(ENDPOINT_BEING_TESTED)
+                    .header(AUTHORIZATION, IDAM_AUTHORIZATION_TOKEN)
+                    .header(SERVICE_AUTHORIZATION, SERVICE_AUTHORIZATION_TOKEN)
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+            ).andExpect(
+                ResultMatcher.matchAll(
+                    status().is5xxServerError(),
+                    content().contentType(APPLICATION_PROBLEM_JSON_VALUE),
+                    jsonPath("$.type").value("https://github.com/hmcts/wa-task-management-api/problem/task-complete-error"),
+                    jsonPath("$.title").value("Task Complete Error"),
+                    jsonPath("$.status").value(500),
+                    jsonPath("$.detail").value(
+                        "Task Complete Error: Task complete partially succeeded. "
+                        + "The Task state was updated to completed, but the Task could not be completed.")
```

Comment:
Does a task need to be reassigned to be completed? If so, Should the string read "The Task state was updated to assigned, but the Task could not be completed."

Reply:
In this particualr case no, because we perform a check:
hasAccessWithAssigneeCheckAndHierarchy

Which does the following checks:
If you're a TCW check if you're assigned to the task and have enough permissions.

Or if you are Senior TCW check that you have enough permissions and complete on behalf of whomever was assigned, this was introduced as part of:
https://github.com/hmcts/wa-task-management-api/pull/258